components/userDetails/userEdit.tsx

```diff
@@ -1,113 +1,136 @@
 import { useMutation } from 'urql'
+
 export interface IFormEdittUser {
-  userEdit: UserProfile
+  userProfileEdit: UserProfile
 }
 
-const UserForm: React.FC<IFormEdittUser> = ({ userEdit }) => {
-  console.log(userEdit.userName)
+const UserForm: React.FC<IFormEdittUser> = ({ userProfileEdit }) => {
+  const [IsCreateUserProfile, setIsCreateUserProfile] = useState(
+    userProfileEdit.id === null || userProfileEdit.id == undefined,
+  )
 
-  /* const  [mutationResult, executeMutation]= useMutation('')
-  if (userEdit?.id.length>0 )
-  {
-    executeMutation( useMutation(`mutation useCreate($input: UserProfileInput!) {
-      createUser(input: $input) {
-        id
-        userName
-      }
-      }`,
-))
-  }
-  else 
-  {
-    executeMutation( useMutation(` updateUser ($id: String!, $input: UserProfileInput!) {
-      updateUser(id: $id, input:$input) {
-        id
-        userName
-      }
-      }`,
-       ))
+  const mutationStringCreate =
+    'mutation useCreate($input: UserProfileInput!) {createUser(input: $input) {id userName }  }'
+  const mutationStringUpdate =
+    'mutation useUpdate($id: String!, $input: UserProfileInput!) {updateUser(id: $id, input:$input) {id  userName}} '
 
-  } */
-  const [mutationResult, executeMutation] =
-    useMutation(`mutation useCreate($input: UserProfileInput!) {
-              createUser(input: $input) {
-                id
-                userName
-              }
-              }`)
+  const [mutationResultCreate, executeMutationCreate] =
+    useMutation(mutationStringCreate)
+  const [mutationResultUpdate, executeMutationUpdate] =
+    useMutation(mutationStringUpdate)
 
-  const { register, handleSubmit, formState } = useForm<UserProfile>({
-    defaultValues: {
-      id: userEdit.id,
-      userName: userEdit.userName,
+  const { register, getValues, handleSubmit, formState } = useForm<UserProfile>(
+    {
+      defaultValues: {
+        id: userProfileEdit.id,
+        userName: userProfileEdit.userName,
+        city: userProfileEdit.city,
+        countryRank: userProfileEdit.countryRank,
+        email: userProfileEdit.email,
+        country: userProfileEdit.country,
+      },
     },
-  })
+  )
+
+  const onSubmit: SubmitHandler<UserProfile> = async (
+    data: Partial<UserProfile>,
+  ) => {
+    if (IsCreateUserProfile) {
+      setIsCreateUserProfile(false)
+      await executeMutationCreate({ input: data })
+    } else {
+      if (data?.id) {
+        delete data.id
+      }
+      const userProfil =
+        userProfileEdit?.id == undefined
+          ? mutationResultCreate.data.createUser.id
+          : userProfileEdit?.id
 
-  const onSubmit: SubmitHandler<UserProfile> = async (data: UserProfile) => {
-    await executeMutation({ input: data })
+      await executeMutationUpdate({ id: userProfil, input: data })
+    }
   }
 
   const errors = formState.errors
-  //const formValues = getValues()
+  const formValues = getValues()
+
+  let erroDiv
```

Comment:
```suggestion
  const errorMessage = IsCreateUserProfile ? mutationResultCreate?.error?.message : mutationResultUpdate?.error?.message
```

...later on render...
```
<div className="text-red-600">{errorMessage}</div>
```